### ISSUE TITLE
v0.47 tests are not passed under strawberry perl x64 v5.26.1

### ISSUE THREAD

**Issue by zdm** [danaj/Math-Prime-Util-GMP#9]
hi, here is build log:

```
cpanm (App::cpanminus) 1.7043 on perl 5.026001 built for MSWin32-x64-multi-thread
Work directory is C:\Users\zdm\AppData\Local\Temp\.cpanm/work/1509915186.12012
You have make D:\devel\perl\c\bin\gmake.exe
You have LWP 6.27
You have D:\devel\msys2\usr\bin\tar.exe, D:\devel\msys2\usr\bin\gzip.exe and D:\devel\msys2\usr\bin\bzip2.exe
--> Working on D/DA/DANAJ/Math-Prime-Util-GMP-0.47.tar.gz
Fetching https://cpan.metacpan.org/authors/id/D/DA/DANAJ/Math-Prime-Util-GMP-0.47.tar.gz ... OK
Unpacking Math-Prime-Util-GMP-0.47.tar.gz
Math-Prime-Util-GMP-0.47/
Math-Prime-Util-GMP-0.47/simpqs.h
Math-Prime-Util-GMP-0.47/utility.c
Math-Prime-Util-GMP-0.47/ecpp.h
Math-Prime-Util-GMP-0.47/squfof126.h
Math-Prime-Util-GMP-0.47/META.json
Math-Prime-Util-GMP-0.47/pbrent63.h
Math-Prime-Util-GMP-0.47/.travis.yml
Math-Prime-Util-GMP-0.47/gmp_main.c
Math-Prime-Util-GMP-0.47/TODO
Math-Prime-Util-GMP-0.47/simpqs.c
Math-Prime-Util-GMP-0.47/factor.h
Math-Prime-Util-GMP-0.47/pbrent63.c
Math-Prime-Util-GMP-0.47/aks.c
Math-Prime-Util-GMP-0.47/ecm.h
Math-Prime-Util-GMP-0.47/utility.h
Math-Prime-Util-GMP-0.47/LICENSE
Math-Prime-Util-GMP-0.47/random_prime.h
Math-Prime-Util-GMP-0.47/README
Math-Prime-Util-GMP-0.47/random_prime.c
Math-Prime-Util-GMP-0.47/isaac.h
Math-Prime-Util-GMP-0.47/t/
Math-Prime-Util-GMP-0.47/t/24-bernfrac.t
Math-Prime-Util-GMP-0.47/t/16-provableprime.t
Math-Prime-Util-GMP-0.47/t/26-lambertw.t
Math-Prime-Util-GMP-0.47/t/28-randomprime.t
Math-Prime-Util-GMP-0.47/t/26-mod.t
Math-Prime-Util-GMP-0.47/t/93-release-spelling.t
Math-Prime-Util-GMP-0.47/t/28-rand.t
Math-Prime-Util-GMP-0.47/t/02-can.t
Math-Prime-Util-GMP-0.47/t/26-combinatorial.t
Math-Prime-Util-GMP-0.47/t/26-riemann.t
Math-Prime-Util-GMP-0.47/t/25-pi.t
Math-Prime-Util-GMP-0.47/t/11-primes.t
Math-Prime-Util-GMP-0.47/t/12-nextprime.t
Math-Prime-Util-GMP-0.47/t/26-logs.t
Math-Prime-Util-GMP-0.47/t/28-urandom.t
Math-Prime-Util-GMP-0.47/t/26-ismisc.t
Math-Prime-Util-GMP-0.47/t/21-conseq-lcm.t
Math-Prime-Util-GMP-0.47/t/91-release-pod-syntax.t
Math-Prime-Util-GMP-0.47/t/26-mersenne.t
Math-Prime-Util-GMP-0.47/t/20-primorial.t
Math-Prime-Util-GMP-0.47/t/90-release-perlcritic.t
Math-Prime-Util-GMP-0.47/t/23-gcd.t
Math-Prime-Util-GMP-0.47/t/17-pseudoprime.t
Math-Prime-Util-GMP-0.47/t/19-moebius.t
Math-Prime-Util-GMP-0.47/t/50-factoring.t
Math-Prime-Util-GMP-0.47/t/13-primecount.t
Math-Prime-Util-GMP-0.47/t/26-roots.t
Math-Prime-Util-GMP-0.47/t/15-probprime.t
Math-Prime-Util-GMP-0.47/t/92-release-pod-coverage.t
Math-Prime-Util-GMP-0.47/t/22-partitions.t
Math-Prime-Util-GMP-0.47/t/27-clusters.t
Math-Prime-Util-GMP-0.47/t/26-digits.t
Math-Prime-Util-GMP-0.47/t/01-load.t
Math-Prime-Util-GMP-0.47/t/10-isprime.t
Math-Prime-Util-GMP-0.47/XS.xs
Math-Prime-Util-GMP-0.47/lib/
Math-Prime-Util-GMP-0.47/lib/Math/
Math-Prime-Util-GMP-0.47/lib/Math/Prime/
Math-Prime-Util-GMP-0.47/lib/Math/Prime/Util/
Math-Prime-Util-GMP-0.47/lib/Math/Prime/Util/GMP.pm
Math-Prime-Util-GMP-0.47/prime_iterator.h
Math-Prime-Util-GMP-0.47/META.yml
Math-Prime-Util-GMP-0.47/ecpp.c
Math-Prime-Util-GMP-0.47/examples/
Math-Prime-Util-GMP-0.47/examples/bench-mp-psrp.pl
Math-Prime-Util-GMP-0.47/examples/vcert.c
Math-Prime-Util-GMP-0.47/examples/verify_primegap.pl
Math-Prime-Util-GMP-0.47/examples/verify-cert.pl
Math-Prime-Util-GMP-0.47/examples/convert-primo-cert.pl
Math-Prime-Util-GMP-0.47/examples/convert-gmpecpp-cert.pl
Math-Prime-Util-GMP-0.47/MANIFEST
Math-Prime-Util-GMP-0.47/class_poly_data.h
Math-Prime-Util-GMP-0.47/bls75.h
Math-Prime-Util-GMP-0.47/aks.h
Math-Prime-Util-GMP-0.47/prime_iterator.c
Math-Prime-Util-GMP-0.47/bls75.c
Math-Prime-Util-GMP-0.47/inc/
Math-Prime-Util-GMP-0.47/inc/Devel/
Math-Prime-Util-GMP-0.47/inc/Devel/CheckLib.pm
Math-Prime-Util-GMP-0.47/primality.c
Math-Prime-Util-GMP-0.47/Makefile.PL
Math-Prime-Util-GMP-0.47/ecm.c
Math-Prime-Util-GMP-0.47/isaac.c
Math-Prime-Util-GMP-0.47/gmp_main.h
Math-Prime-Util-GMP-0.47/ptypes.h
Math-Prime-Util-GMP-0.47/squfof126.c
Math-Prime-Util-GMP-0.47/xt/
Math-Prime-Util-GMP-0.47/xt/expr.c
Math-Prime-Util-GMP-0.47/xt/bench-random-bytes.pl
Math-Prime-Util-GMP-0.47/xt/expr.h
Math-Prime-Util-GMP-0.47/xt/llr.pl
Math-Prime-Util-GMP-0.47/xt/proof-text-format.txt
Math-Prime-Util-GMP-0.47/xt/calculate-mr-probs.pl
Math-Prime-Util-GMP-0.47/xt/create-standalone.sh
Math-Prime-Util-GMP-0.47/xt/expr-impl.h
Math-Prime-Util-GMP-0.47/Changes
Math-Prime-Util-GMP-0.47/primality.h
Math-Prime-Util-GMP-0.47/factor.c
Entering Math-Prime-Util-GMP-0.47
Checking configure dependencies from META.json
Checking if you have ExtUtils::MakeMaker 6.58 ... Yes (7.30)
Running Makefile.PL
Configuring Math-Prime-Util-GMP-0.47 ... Checking if your kit is complete...
Looks good
Generating a gmake-style Makefile
Writing Makefile for Math::Prime::Util::GMP
Writing MYMETA.yml and MYMETA.json
OK
Checking dependencies from MYMETA.json ...
Checking if you have Exporter 5.57 ... Yes (5.72)
Checking if you have ExtUtils::MakeMaker 0 ... Yes (7.30)
Checking if you have base 0 ... Yes (2.26)
Checking if you have Carp 0 ... Yes (1.42)
Checking if you have Math::BigInt 1.88 ... Yes (1.999811)
Checking if you have XSLoader 0.01 ... Yes (0.27)
Checking if you have Fcntl 0 ... Yes (1.13)
Checking if you have Test::More 0.45 ... Yes (1.302106)
Building and testing Math-Prime-Util-GMP-0.47 ... cp lib/Math/Prime/Util/GMP.pm blib\lib\Math\Prime\Util\GMP.pm
Running Mkbootstrap for GMP ()
"D:\devel\perl\perl\bin\perl.exe" -MExtUtils::Command -e chmod -- 644 "GMP.bs"
"D:\devel\perl\perl\bin\perl.exe" -MExtUtils::Command::MM -e cp_nonempty -- GMP.bs blib\arch\auto\Math\Prime\Util\GMP\GMP.bs 644
gcc -c   -s -O2 -DWIN32 -DWIN64 -DCONSERVATIVE -D__USE_MINGW_ANSI_STDIO -DPERL_TEXTMODE_SCRIPTS -DPERL_IMPLICIT_CONTEXT -DPERL_IMPLICIT_SYS -DUSE_PERLIO -fwrapv -fno-strict-aliasing -mms-bitfields -s -O2   -DVERSION=\"0.47\" -DXS_VERSION=\"0.47\"  "-ID:\devel\perl\perl\lib\CORE"   prime_iterator.c
gcc -c   -s -O2 -DWIN32 -DWIN64 -DCONSERVATIVE -D__USE_MINGW_ANSI_STDIO -DPERL_TEXTMODE_SCRIPTS -DPERL_IMPLICIT_CONTEXT -DPERL_IMPLICIT_SYS -DUSE_PERLIO -fwrapv -fno-strict-aliasing -mms-bitfields -s -O2   -DVERSION=\"0.47\" -DXS_VERSION=\"0.47\"  "-ID:\devel\perl\perl\lib\CORE"   utility.c
gcc -c   -s -O2 -DWIN32 -DWIN64 -DCONSERVATIVE -D__USE_MINGW_ANSI_STDIO -DPERL_TEXTMODE_SCRIPTS -DPERL_IMPLICIT_CONTEXT -DPERL_IMPLICIT_SYS -DUSE_PERLIO -fwrapv -fno-strict-aliasing -mms-bitfields -s -O2   -DVERSION=\"0.47\" -DXS_VERSION=\"0.47\"  "-ID:\devel\perl\perl\lib\CORE"   primality.c
gcc -c   -s -O2 -DWIN32 -DWIN64 -DCONSERVATIVE -D__USE_MINGW_ANSI_STDIO -DPERL_TEXTMODE_SCRIPTS -DPERL_IMPLICIT_CONTEXT -DPERL_IMPLICIT_SYS -DUSE_PERLIO -fwrapv -fno-strict-aliasing -mms-bitfields -s -O2   -DVERSION=\"0.47\" -DXS_VERSION=\"0.47\"  "-ID:\devel\perl\perl\lib\CORE"   factor.c
gcc -c   -s -O2 -DWIN32 -DWIN64 -DCONSERVATIVE -D__USE_MINGW_ANSI_STDIO -DPERL_TEXTMODE_SCRIPTS -DPERL_IMPLICIT_CONTEXT -DPERL_IMPLICIT_SYS -DUSE_PERLIO -fwrapv -fno-strict-aliasing -mms-bitfields -s -O2   -DVERSION=\"0.47\" -DXS_VERSION=\"0.47\"  "-ID:\devel\perl\perl\lib\CORE"   pbrent63.c
gcc -c   -s -O2 -DWIN32 -DWIN64 -DCONSERVATIVE -D__USE_MINGW_ANSI_STDIO -DPERL_TEXTMODE_SCRIPTS -DPERL_IMPLICIT_CONTEXT -DPERL_IMPLICIT_SYS -DUSE_PERLIO -fwrapv -fno-strict-aliasing -mms-bitfields -s -O2   -DVERSION=\"0.47\" -DXS_VERSION=\"0.47\"  "-ID:\devel\perl\perl\lib\CORE"   squfof126.c
gcc -c   -s -O2 -DWIN32 -DWIN64 -DCONSERVATIVE -D__USE_MINGW_ANSI_STDIO -DPERL_TEXTMODE_SCRIPTS -DPERL_IMPLICIT_CONTEXT -DPERL_IMPLICIT_SYS -DUSE_PERLIO -fwrapv -fno-strict-aliasing -mms-bitfields -s -O2   -DVERSION=\"0.47\" -DXS_VERSION=\"0.47\"  "-ID:\devel\perl\perl\lib\CORE"   simpqs.c
gcc -c   -s -O2 -DWIN32 -DWIN64 -DCONSERVATIVE -D__USE_MINGW_ANSI_STDIO -DPERL_TEXTMODE_SCRIPTS -DPERL_IMPLICIT_CONTEXT -DPERL_IMPLICIT_SYS -DUSE_PERLIO -fwrapv -fno-strict-aliasing -mms-bitfields -s -O2   -DVERSION=\"0.47\" -DXS_VERSION=\"0.47\"  "-ID:\devel\perl\perl\lib\CORE"   ecm.c
gcc -c   -s -O2 -DWIN32 -DWIN64 -DCONSERVATIVE -D__USE_MINGW_ANSI_STDIO -DPERL_TEXTMODE_SCRIPTS -DPERL_IMPLICIT_CONTEXT -DPERL_IMPLICIT_SYS -DUSE_PERLIO -fwrapv -fno-strict-aliasing -mms-bitfields -s -O2   -DVERSION=\"0.47\" -DXS_VERSION=\"0.47\"  "-ID:\devel\perl\perl\lib\CORE"   bls75.c
gcc -c   -s -O2 -DWIN32 -DWIN64 -DCONSERVATIVE -D__USE_MINGW_ANSI_STDIO -DPERL_TEXTMODE_SCRIPTS -DPERL_IMPLICIT_CONTEXT -DPERL_IMPLICIT_SYS -DUSE_PERLIO -fwrapv -fno-strict-aliasing -mms-bitfields -s -O2   -DVERSION=\"0.47\" -DXS_VERSION=\"0.47\"  "-ID:\devel\perl\perl\lib\CORE"   ecpp.c
gcc -c   -s -O2 -DWIN32 -DWIN64 -DCONSERVATIVE -D__USE_MINGW_ANSI_STDIO -DPERL_TEXTMODE_SCRIPTS -DPERL_IMPLICIT_CONTEXT -DPERL_IMPLICIT_SYS -DUSE_PERLIO -fwrapv -fno-strict-aliasing -mms-bitfields -s -O2   -DVERSION=\"0.47\" -DXS_VERSION=\"0.47\"  "-ID:\devel\perl\perl\lib\CORE"   aks.c
gcc -c   -s -O2 -DWIN32 -DWIN64 -DCONSERVATIVE -D__USE_MINGW_ANSI_STDIO -DPERL_TEXTMODE_SCRIPTS -DPERL_IMPLICIT_CONTEXT -DPERL_IMPLICIT_SYS -DUSE_PERLIO -fwrapv -fno-strict-aliasing -mms-bitfields -s -O2   -DVERSION=\"0.47\" -DXS_VERSION=\"0.47\"  "-ID:\devel\perl\perl\lib\CORE"   gmp_main.c
gcc -c   -s -O2 -DWIN32 -DWIN64 -DCONSERVATIVE -D__USE_MINGW_ANSI_STDIO -DPERL_TEXTMODE_SCRIPTS -DPERL_IMPLICIT_CONTEXT -DPERL_IMPLICIT_SYS -DUSE_PERLIO -fwrapv -fno-strict-aliasing -mms-bitfields -s -O2   -DVERSION=\"0.47\" -DXS_VERSION=\"0.47\"  "-ID:\devel\perl\perl\lib\CORE"   isaac.c
gcc -c   -s -O2 -DWIN32 -DWIN64 -DCONSERVATIVE -D__USE_MINGW_ANSI_STDIO -DPERL_TEXTMODE_SCRIPTS -DPERL_IMPLICIT_CONTEXT -DPERL_IMPLICIT_SYS -DUSE_PERLIO -fwrapv -fno-strict-aliasing -mms-bitfields -s -O2   -DVERSION=\"0.47\" -DXS_VERSION=\"0.47\"  "-ID:\devel\perl\perl\lib\CORE"   random_prime.c
"D:\devel\perl\perl\bin\perl.exe" "D:\devel\perl\perl\lib\ExtUtils/xsubpp"  -typemap D:\devel\perl\perl\lib\ExtUtils\typemap  XS.xs > XS.xsc
"D:\devel\perl\perl\bin\perl.exe" -MExtUtils::Command -e mv -- XS.xsc XS.c
gcc -c   -s -O2 -DWIN32 -DWIN64 -DCONSERVATIVE -D__USE_MINGW_ANSI_STDIO -DPERL_TEXTMODE_SCRIPTS -DPERL_IMPLICIT_CONTEXT -DPERL_IMPLICIT_SYS -DUSE_PERLIO -fwrapv -fno-strict-aliasing -mms-bitfields -s -O2   -DVERSION=\"0.47\" -DXS_VERSION=\"0.47\"  "-ID:\devel\perl\perl\lib\CORE"   XS.c
"D:\devel\perl\perl\bin\perl.exe" -MExtUtils::Mksymlists \
     -e "Mksymlists('NAME'=>\"Math::Prime::Util::GMP\", 'DLBASE' => 'GMP', 'DL_FUNCS' => {  }, 'FUNCLIST' => [], 'IMPORTS' => {  }, 'DL_VARS' => []);"
g++.exe GMP.def -o blib\arch\auto\Math\Prime\Util\GMP\GMP.xs.dll -mdll -s -L"D:\devel\perl\perl\lib\CORE" -L"D:\devel\perl\c\lib" prime_iterator.o utility.o primality.o factor.o pbrent63.o squfof126.o simpqs.o ecm.o bls75.o ecpp.o aks.o gmp_main.o isaac.o random_prime.o XS.o   "D:\devel\perl\perl\lib\CORE\libperl526.a" "D:\devel\perl\c\lib\libgmp.a" "D:\devel\perl\c\x86_64-w64-mingw32\lib\libm.a" "D:\devel\perl\c\x86_64-w64-mingw32\lib\libmoldname.a" "D:\devel\perl\c\x86_64-w64-mingw32\lib\libkernel32.a" "D:\devel\perl\c\x86_64-w64-mingw32\lib\libuser32.a" "D:\devel\perl\c\x86_64-w64-mingw32\lib\libgdi32.a" "D:\devel\perl\c\x86_64-w64-mingw32\lib\libwinspool.a" "D:\devel\perl\c\x86_64-w64-mingw32\lib\libcomdlg32.a" "D:\devel\perl\c\x86_64-w64-mingw32\lib\libadvapi32.a" "D:\devel\perl\c\x86_64-w64-mingw32\lib\libshell32.a" "D:\devel\perl\c\x86_64-w64-mingw32\lib\libole32.a" "D:\devel\perl\c\x86_64-w64-mingw32\lib\liboleaut32.a" "D:\devel\perl\c\x86_64-w64-mingw32\lib\libnetapi32.a" "D:\devel\perl\c\x86_64-w64-mingw32\lib\libuuid.a" "D:\devel\perl\c\x86_64-w64-mingw32\lib\libws2_32.a" "D:\devel\perl\c\x86_64-w64-mingw32\lib\libmpr.a" "D:\devel\perl\c\x86_64-w64-mingw32\lib\libwinmm.a" "D:\devel\perl\c\x86_64-w64-mingw32\lib\libversion.a" "D:\devel\perl\c\x86_64-w64-mingw32\lib\libodbc32.a" "D:\devel\perl\c\x86_64-w64-mingw32\lib\libodbccp32.a" "D:\devel\perl\c\x86_64-w64-mingw32\lib\libcomctl32.a" -Wl,--enable-auto-image-base
"D:\devel\perl\perl\bin\perl.exe" -MExtUtils::Command -e chmod -- 755 blib\arch\auto\Math\Prime\Util\GMP\GMP.xs.dll
"D:\devel\perl\perl\bin\perl.exe" -MExtUtils::Command::MM -e cp_nonempty -- GMP.bs blib\arch\auto\Math\Prime\Util\GMP\GMP.bs 644
"D:\devel\perl\perl\bin\perl.exe" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib\lib', 'blib\arch')" t/*.t
t/01-load.t .................. ok
t/02-can.t ................... ok
t/10-isprime.t ............... ok
t/11-primes.t ................ ok
t/12-nextprime.t ............. ok
t/13-primecount.t ............ ok
t/15-probprime.t ............. ok
t/16-provableprime.t ......... 1/138 Math::Prime::Util internal error: bad n in pbrent_factor at C:\Users\zdm\AppData\Local\Temp\.cpanm\work\1509915186.12012\Math-Prime-Util-GMP-0.47\blib\lib/Math/Prime/Util/GMP.pm line 155.
# Looks like your test exited with 255 just after 102.
t/16-provableprime.t ......... Dubious, test returned 255 (wstat 65280, 0xff00)
Failed 36/138 subtests
t/17-pseudoprime.t ........... ok
t/19-moebius.t ............... 1/191 Incorrect factoring at t/19-moebius.t line 247.
# Looks like your test exited with 255 just after 120.
t/19-moebius.t ............... Dubious, test returned 255 (wstat 65280, 0xff00)
Failed 71/191 subtests
t/20-primorial.t ............. ok
t/21-conseq-lcm.t ............ ok
t/22-partitions.t ............ ok
t/23-gcd.t ................... ok
t/24-bernfrac.t .............. ok
t/25-pi.t .................... ok
t/26-combinatorial.t ......... ok
t/26-digits.t ................ ok
t/26-ismisc.t ................ 1/19 Incorrect factoring at t/26-ismisc.t line 30.
# Looks like your test exited with 255 just after 5.
t/26-ismisc.t ................ Dubious, test returned 255 (wstat 65280, 0xff00)
Failed 14/19 subtests
t/26-lambertw.t .............. ok
t/26-logs.t .................. ok
t/26-mersenne.t .............. ok
t/26-mod.t ................... ok
t/26-riemann.t ............... ok
t/26-roots.t ................. ok
t/27-clusters.t .............. ok
t/28-rand.t .................. ok
t/28-randomprime.t ........... ok
t/28-urandom.t ............... ok
t/50-factoring.t ............. 1/164 Incorrect factoring at t/50-factoring.t line 113.
# Looks like your test exited with 255 just after 65.
t/50-factoring.t ............. Dubious, test returned 255 (wstat 65280, 0xff00)
Failed 99/164 subtests
t/90-release-perlcritic.t .... skipped: these tests are for release candidate testing
t/91-release-pod-syntax.t .... skipped: these tests are for release candidate testing
t/92-release-pod-coverage.t .. skipped: these tests are for release candidate testing
t/93-release-spelling.t ...... skipped: these tests are for release candidate testing

Test Summary Report
-------------------
t/16-provableprime.t       (Wstat: 65280 Tests: 102 Failed: 0)
  Non-zero exit status: 255
  Parse errors: Bad plan.  You planned 138 tests but ran 102.
t/19-moebius.t             (Wstat: 65280 Tests: 120 Failed: 0)
  Non-zero exit status: 255
  Parse errors: Bad plan.  You planned 191 tests but ran 120.
t/26-ismisc.t              (Wstat: 65280 Tests: 5 Failed: 0)
  Non-zero exit status: 255
  Parse errors: Bad plan.  You planned 19 tests but ran 5.
t/50-factoring.t           (Wstat: 65280 Tests: 65 Failed: 0)
  Non-zero exit status: 255
  Parse errors: Bad plan.  You planned 164 tests but ran 65.
Files=34, Tests=2713, 10 wallclock secs ( 0.39 usr +  0.09 sys =  0.48 CPU)
Result: FAIL
Failed 4/34 test programs. 0/2713 subtests failed.
gmake: *** [Makefile:1115: test_dynamic] Error 255
FAIL
! Installing D/DA/DANAJ/Math-Prime-Util-GMP-0.47.tar.gz failed. See C:\Users\zdm\AppData\Local\Temp\.cpanm\work\1509915186.12012\build.log for details. Retry with --force to force install it.
```

**Comment by danaj** [comment]
Fixed in 0.48.  This happens with a 64-bit Perl linked with a 32-bit GMP library.

Thanks for report.  Could you let me know if it is fixed?

**Comment by zdm** [comment]
yes, 0.48 installed without errors.
thank you.
